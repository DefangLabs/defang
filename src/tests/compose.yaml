name: tests
services:
  dfnx:
    build:
      context: .
      dockerfile: Dockerfile
      args:
        DNS: "${COMPOSE_PROJECT_NAME}-dfnx.prod1.defang.dev"
    env_file:
      - fileName.env
    deploy:
      resources:
        limits:
          cpus: '0.50'
          memory: 512M
        reservations:
          cpus: '0.25'
          memory: 256M
    ports:
      - target: 80
        mode: ingress
      - target: 1234
        # mode: host
    secrets:
      - dummy
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
      # disable: true

  # dfnx:
  #   build:
  #     context: .
      # dockerfile: Dockerfile.dfn
    # ports:
      # - 80

  echo:
    image: ealen/echo-server
    ports:
      - target: 80
        mode: ingress
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
    # domainname: echotest.gnafed.click
    profiles:
      - donotstart

secrets:
  dummy:
    external: true
    name: dummy
