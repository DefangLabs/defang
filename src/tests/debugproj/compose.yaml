services:
  failing:
    build: ./app
  ok:
    build: .
