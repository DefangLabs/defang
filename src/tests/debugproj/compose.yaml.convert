[
  {
    "name": "failing",
    "platform": 2,
    "internal": true,
    "build": {
      "context": "./app",
      "dockerfile": "Dockerfile"
    },
    "networks": 1
  },
  {
    "name": "ok",
    "platform": 2,
    "internal": true,
    "build": {
      "context": ".",
      "dockerfile": "Dockerfile"
    },
    "networks": 1
  }
]