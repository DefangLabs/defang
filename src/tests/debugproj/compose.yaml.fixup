[
  {
    "build": {
      "context": "./app",
      "dockerfile": "Dockerfile"
    },
    "command": null,
    "entrypoint": null,
    "networks": {
      "default": null
    }
  },
  {
    "build": {
      "context": ".",
      "dockerfile": "Dockerfile"
    },
    "command": null,
    "entrypoint": null,
    "networks": {
      "default": null
    }
  }
]