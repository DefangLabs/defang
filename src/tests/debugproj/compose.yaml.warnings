 ! service "failing": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)
 ! service "failing": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "ok": missing compose directive: restart; assuming 'unless-stopped' (add 'restart' to silence)
 ! service "ok": missing memory reservation; specify deploy.resources.reservations.memory to avoid out-of-memory errors
 * Compressing build context for failing at ./app
 * Compressing build context for ok at .