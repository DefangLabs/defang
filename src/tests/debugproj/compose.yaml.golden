name: debugproj
services:
  failing:
    build:
      context: ./app
      dockerfile: Dockerfile
    networks:
      default: null
  ok:
    build:
      context: .
      dockerfile: Dockerfile
    networks:
      default: null
networks:
  default:
    name: debugproj_default
